src/web/mjs/connectors/SoftKomik.mjs

```diff
@@ -27,16 +27,16 @@ export default class SoftKomik extends Connector {
         let mangaList = [];
         for (let page = 1, run = true; run; page++) {
             const mangas = await this._getMangasFromPage(page);
-            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+            mangas.length > 27 ? mangaList.push(...mangas) : run = false;
```

Comment:
That means that if there are 26 titles on the page they won't get retrieved. I think there's an issue here :)

Reply:
My bad, fix the issue alr

Reply:
My bad, fix the issue alr

Reply:
why did you remove the ternary operator (if/else) ?
If the list length is 0 (when reaching the last page and the preview page was full with 26 items) it would push the undefined result into the array.
+
lint error ;)

Reply:
The api will keep giving the same result when it reach the last page. For example the last page have 26 data, last page +1 will also have 26 and not 0

Reply:
Will this be a better way to write the code?
`for (let page = 1, run = true; run; page++) {
            const mangas = await this._getMangasFromPage(page);
            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
            mangas.length > 27 ? mangaList.push(...mangas) : run = false;
}
mangaList.push(..mangas)
`

Reply:
I think we need to compare the last results to find if thei're equals and then get out of the loop.
27 is not a value that should be used as on my side i get 56 results per page
https://mongo.animeyusha.com/komik?page=50

Reply:
I think we can leave it like that (your last code in comments). If someone requests a few missing titles we'll take a look for an alternative as we're not able to unit test  of what happens when the last page has 28 titles.

Reply:
I feel that this code will be better
```    async _getMangas() {
        let mangaList = [];
        for (let page = 1, run = true; run; page++) {
            const mangas = await this._getMangasFromPage(page);
            if(mangas.length == 0){
                run = false;
            }
            else if (mangaList.length == 0 || mangas[mangas.length - 1].id !== mangaList[mangaList.length - 1].id){
                mangaList.push(...mangas);
            }
            else{
                run = false;
            }
        }
        return mangaList;
    }
```